### PR TITLE
Rbd_utils_refactor

### PR DESCRIPTION
**Pull request by HaruChebrolu**
Signed-off-by: Chebrolu Harika <hchebrol@redhat.com>

# Description

lib folder contains all SDK calls for RBD
tests/rbd/test_configs contains testcase specific workflows

Scenario covered: Snap and clone operations on imported image
Conf file used: conf/pacific/rbd/tier-0_rbd.yaml
Command used: python run.py --osp-cred osp/osp-cred-ci-2.yaml --log-level debug --build latest --inventory conf/inventory/rhel-8-latest.yaml --global-conf conf/pacific/rbd/tier-0_rbd.yaml --suite suites/pacific/rbd/tier-2_rbd_mirroring.yaml --rhbuild 5.1 --platform rhel-8

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

File: ceph/rbd/snapshot.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
+from ceph.rbd.rbd import Rbd
+from tests.rbd.exceptions import CreateCloneError, ProtectSnapError, SnapCreateError
+from utility.log import Log
+
+log = Log(__name__)
+
+
+class RbdSnapShot(Rbd):
```
Comment (psathyan) [inline ceph/rbd/snapshot.py:8]: Would it make sense to have the object/class name as `Snapshot`

File: suites/pacific/rbd/tier-2_rbd_mirroring.yaml [hunk 2]
```diff
@@ -0,0 +1,83 @@
+# Tier2: Extended RBD testing
```
Comment (psathyan) [inline suites/pacific/rbd/tier-2_rbd_mirroring.yaml:1]: We recommend to have the file name as `tier-2_rbd_mirror_test-basic-features-of-mirroring.yaml`. This allows to get metadata information from the filename.

File: tests/rbd/test_rbdV2.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        log.info(method)
+        method(step_config["args"])
+    return 0
+
+
+def run(**kw):
+    """
+    Rbd Workflow module to manage ceph-rbd services
+
+    Arguments:
```
Comment (psathyan) [inline tests/rbd/test_rbdV2.py:58]: Could we have an example on one should call/use this module?

File: ceph/rbd/rbd.py [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+                    f"sleeping for 2 seconds and checking status again"
+                )
+                sleep(2)
+            else:
+                log.info(f"pool {pool_name} exists in the cluster")
+                return True
+        log.info(f"pool {pool_name} does not exist on cluster")
+        return False
+
+    def exec_cmd(self, **kw):
```
Comment (psathyan) [inline ceph/rbd/rbd.py:45]: The return value are not consistent. Could we revisit it?

File: ceph/rbd/snapshot.py (hunk 1 above)
Comment (HaruChebrolu) [reply ceph/rbd/snapshot.py]: done, renamed to Snapshot

File: suites/pacific/rbd/tier-2_rbd_mirroring.yaml (hunk 2 above)
Comment (HaruChebrolu) [reply suites/pacific/rbd/tier-2_rbd_mirroring.yaml]: done

File: ceph/rbd/rbd.py (hunk 4 above)
Comment (HaruChebrolu) [reply ceph/rbd/rbd.py]: done

**Review by psathyan — APPROVED** [review]
Let's get this in and we continue to improve on it

**Pull request merged**